File: samples/java/TicTacToe/Client/src/main/java/systems/zlink/samples/tictactoe/client/TicTacToeClientOptions.java
package systems.zlink.samples.tictactoe.client;

public record TicTacToeClientOptions(
        String apiUrl,
        String gameName,
        String xActorId,
        String oActorId,
        String observerActorId,
        String lifecycleCompletionFile) {
    public static TicTacToeClientOptions createDefault() {
        return new TicTacToeClientOptions(
                TicTacToeSampleDefaults.ApiUrl,
                TicTacToeSampleDefaults.GameName,
                TicTacToeSampleDefaults.XActorId,
                TicTacToeSampleDefaults.OActorId,
                TicTacToeSampleDefaults.ObserverActorId,
                "");
    }
}

File: samples/kotlin/TicTacToe/Client/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/client/TicTacToeClientOptions.kt
package systems.zlink.samples.kotlin.tictactoe.client

data class TicTacToeClientOptions(
    val apiUrl: String,
    val gameName: String,
    val xActorId: String,
    val oActorId: String,
    val observerActorId: String,
    val lifecycleCompletionFile: String,
) {
    companion object {
        fun createDefault(): TicTacToeClientOptions =
            TicTacToeClientOptions(
                apiUrl = TicTacToeSampleDefaults.ApiUrl,
                gameName = TicTacToeSampleDefaults.GameName,
                xActorId = TicTacToeSampleDefaults.XActorId,
                oActorId = TicTacToeSampleDefaults.OActorId,
                observerActorId = TicTacToeSampleDefaults.ObserverActorId,
                lifecycleCompletionFile = "",
            )
    }
}
